src/odemis/driver/technolution.py

```diff
@@ -1131,9 +1131,17 @@ def _acquire(self):
                         logging.warning("ASM acquisition already had the '%s', received this command again." % command)
                         continue
 
-                    acquisition_in_progress = True
                     megafield_metadata = args[0]
-                    self.parent.asmApiPostCall("/scan/start_mega_field", 204, megafield_metadata.to_dict())
+                    notifier_func = args[1]  # Return function (usually, dataflow.notify or acquire_single_field queue)
+                    try:
+                        self.parent.asmApiPostCall("/scan/start_mega_field", 204, megafield_metadata.to_dict())
+                        notifier_func(None)  # return None if all went fine
+                    except Exception as ex:
+                        logging.error("During the start of the acquisition an error has occurred: %s.", ex)
+                        notifier_func(ex)
+                        continue  # let the caller decide on what to do next
+
+                    acquisition_in_progress = True  # acquisition was successfully started
```

Comment:
If an error is found the var acquisition_in_progress is set to True while actually the acquisition isn't really started/in progress. That seems a bit odd.

Reply:
If an exception is raised, the continue will be executed and this line is not reached. So the variable `acquisition_in_progress` will still be false. Only if the start succeeded (so the try statement was successfully executed) this line will be reached.